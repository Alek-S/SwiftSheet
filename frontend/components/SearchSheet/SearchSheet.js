import React, { useState } from 'react';
import styled from 'styled-components';
import { Card } from '../../defaultStyle';
import { Link } from 'react-router-dom';

const SearchSheet = () => {
	const [value, setValue] = useState('');
	const handleChange = e => {
		setValue(e.target.value);
	};

	return (
		<StyledCard>
			<StyledForm>
				<input
					type="text"
					placeholder="search by sheet ID"
					value={value}
					onChange={handleChange}
				/>
				<Link to={`/${value}`}>
					<SubmitButton>Submit</SubmitButton>
				</Link>
			</StyledForm>
		</StyledCard>
	);
};

const StyledCard = styled(Card)`
	overflow: hidden;
	width: 90%;
	max-width: 600px;
	margin-bottom: 3rem;

	header {
		background: ${props => props.theme.gradient.greenBlue};
		padding-top: 1rem;
		padding-bottom: 1rem;
		width: 100%;
	}

	h2 {
		font-weight: 400;
		text-align: center;
		color: white;
		font-size: 1.1rem;
	}

	input {
		box-sizing: border-box;
		flex-grow: 2;
		height: 30px;
		margin-bottom: 0;
		padding-left: 2rem;
		margin-right: 1rem;
		font-size: 0.95rem;

		::placeholder {
			color: ${props => props.theme.color.backgroundDarkest};
		}
	}

	@media (max-width: 500px) {
		width: 95%;

		input {
			padding-left: 1rem;
			margin-right: 0;
		}
	}
`;

const StyledForm = styled.form`
	margin: 2rem;
	display: flex;
	flex-direction: row;
	flex-wrap: nowrap;
	justify-content: space-between;

	@media (max-width: 500px) {
		flex-direction: column;
		justify-content: center;
	}
`;

const SubmitButton = styled.button`
	outline: none;
	box-sizing: border-box;
	height: 30px;
	width: 100px;
	border-radius: 20px;
	background: ${props => props.theme.gradient.greenBlue};
	border: none;
	color: white;
	font-family: ${props => props.theme.font.main};
	font-size: 1rem;
	transition: all 0.5s;

	&:hover {
		box-shadow: ${props => props.theme.boxShadow};
	}
	@media (max-width: 500px) {
		width: 100%;
		margin: auto;
		margin-top: 2.5rem;
	}
`;

export default SearchSheet;
