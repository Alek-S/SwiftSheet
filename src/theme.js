/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
