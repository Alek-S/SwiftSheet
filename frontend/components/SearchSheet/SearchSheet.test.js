import React from 'react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

import SearchSheet from './SearchSheet';
import theme from '../../theme';

describe('SearchSheet', () => {
	test('matches the snapshot', () => {
		const tree = mount(
			<Router>
				<ThemeProvider theme={theme}>
					<SearchSheet />
				</ThemeProvider>
			</Router>
		);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
