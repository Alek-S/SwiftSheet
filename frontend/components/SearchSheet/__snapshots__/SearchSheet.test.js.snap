// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSheet matches the snapshot 1`] = `
.c0 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
  border-radius: 8px;
  box-shadow: 0px 0px 30px -5px rgba(0,0,0,0.15);
  overflow: hidden;
  width: 90%;
  max-width: 600px;
  margin-bottom: 3rem;
}

.c0 input {
  outline: none;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  background-color: #F1F3F4;
  border: 1px solid #F1F3F4;
  border-radius: 5rem;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 input::-webkit-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input::-moz-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input:-ms-input-placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input::placeholder {
  font-size: 1rem;
  font-weight: 400;
  font-style: italic;
  font-family: 'Quicksand',sans-serif;
  color: #848484;
}

.c0 input:hover {
  border: solid 1px #d9d9d9;
}

.c0 input:focus,
.c0 input:active {
  border: solid 1px #46C8F5;
}

.c0 input.true {
  border: 1px solid #ED5050;
}

.c0 header {
  background: linear-gradient(160deg,#57DACB 0%,#55A2ED 100%);
  padding-top: 1rem;
  padding-bottom: 1rem;
  width: 100%;
}

.c0 h2 {
  font-weight: 400;
  text-align: center;
  color: white;
  font-size: 1.1rem;
}

.c0 input {
  box-sizing: border-box;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  height: 30px;
  margin-bottom: 0;
  padding-left: 2rem;
  margin-right: 1rem;
  font-size: 0.95rem;
}

.c0 input::-webkit-input-placeholder {
  color: #BCBCBC;
}

.c0 input::-moz-placeholder {
  color: #BCBCBC;
}

.c0 input:-ms-input-placeholder {
  color: #BCBCBC;
}

.c0 input::placeholder {
  color: #BCBCBC;
}

.c1 {
  margin: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  height: 30px;
  width: 100px !important;
  border-radius: 20px;
  background: linear-gradient(160deg,#57DACB 0%,#55A2ED 100%);
  border: none;
  color: white !important;
  font-family: 'Quicksand',sans-serif;
  font-size: 1rem;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c2:hover {
  box-shadow: 0px 5px 40px -5px rgba(0,0,0,0.3);
}

@media (max-width:500px) {
  .c0 {
    width: 95%;
  }

  .c0 input {
    padding-left: 1rem;
    margin-right: 0;
  }
}

@media (max-width:500px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:500px) {
  .c2 {
    width: 100%;
    margin: auto;
    margin-top: 2.5rem;
  }
}

<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "boxShadow": "0px 5px 40px -5px rgba(0,0,0,0.3)",
          "boxShadowDark": "0px 5px 40px -2px rgba(0,0,0,0.5)",
          "boxShadowLight": "0px 0px 30px -5px rgba(0,0,0,0.15)",
          "color": Object {
            "aqua": "#68F6BA",
            "background": "#fafafa",
            "backgroundDark": "#EAEAEA",
            "backgroundDarkest": "#BCBCBC",
            "blue": "#46C8F5",
            "border": "#d9d9d9",
            "darkBlue": "#2E8AA8",
            "green": "#D8E13F",
            "input": "#F1F3F4",
            "lightBlue": "#97E4FF",
            "lightRed": "#FF8D8D",
            "lightText": "#848484",
            "orange": "#ffbf80",
            "red": "#ED5050",
            "text": "#404040",
            "vividGreen": "#E3FC52",
          },
          "font": Object {
            "header": "'Poiret One', cursive",
            "main": "'Quicksand', sans-serif",
          },
          "gradient": Object {
            "green": "linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)",
            "greenBlue": " linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)",
            "light": "linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)",
            "lightRed": "linear-gradient(to bottom, #f39191 0%, #ef6262 100%)",
            "red": "linear-gradient(to bottom, #f07878 0%, #ED5050 100%)",
          },
        }
      }
    >
      <SearchSheet>
        <defaultStyle__FormCard>
          <div
            className="defaultStyle-g21vfe-0 defaultStyle__Card-g21vfe-1 c0"
          >
            <defaultStyle__StyledForm>
              <form
                className="c1"
              >
                <input
                  aria-label="input search id"
                  onChange={[Function]}
                  placeholder="search by sheet ID"
                  type="text"
                  value=""
                />
                <Link
                  to="/"
                >
                  <LinkAnchor
                    href="/"
                    navigate={[Function]}
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <defaultStyle__SubmitButton>
                        <button
                          className="c2"
                        >
                          Submit
                        </button>
                      </defaultStyle__SubmitButton>
                    </a>
                  </LinkAnchor>
                </Link>
              </form>
            </defaultStyle__StyledForm>
          </div>
        </defaultStyle__FormCard>
      </SearchSheet>
    </ThemeProvider>
  </Router>
</BrowserRouter>
`;
